packages/components/src/color-picker/inputs.js

```diff
@@ -72,109 +64,100 @@ export class Input extends Component {
+			onChange={ ( newValue ) => handleChange( newValue ) }
+			onBlur={ handleBlur }
+			onKeyDown={ handleKeyDown }
+			{ ...props }
+		/>
+	);
 }
 
-const PureButton = pure( Button );
-
-export class Inputs extends Component {
-	constructor( { hsl } ) {
-		super( ...arguments );
-
-		const view = hsl.a === 1 ? 'hex' : 'rgb';
-		this.state = { view };
+function normalizeValue( valueKey, value ) {
+	if ( valueKey !== 'a' ) {
+		return value;
+	}
 
-		this.toggleViews = this.toggleViews.bind( this );
-		this.resetDraftValues = this.resetDraftValues.bind( this );
-		this.handleChange = this.handleChange.bind( this );
-		this.normalizeValue = this.normalizeValue.bind( this );
+	if ( value < 0 ) {
+		return 0;
+	} else if ( value > 1 ) {
+		return 1;
 	}
+	return Math.round( value * 100 ) / 100;
+}
 
-	static getDerivedStateFromProps( props, state ) {
-		if ( props.hsl.a !== 1 && state.view === 'hex' ) {
-			return { view: 'rgb' };
-		}
-		return null;
+export default function Inputs( {
+	hex,
+	rgb,
+	hsl,
+	onChange,
+	disableAlpha = false,
+} ) {
+	const [ view, setView ] = useState( hsl.a === 1 ? 'hex' : 'rgb' );
+
+	if ( hsl.a !== 1 && view === 'hex' ) {
+		setView( 'rgb' );
 	}
 
-	toggleViews() {
-		if ( this.state.view === 'hex' ) {
-			this.setState( { view: 'rgb' }, this.resetDraftValues );
+	function toggleViews() {
+		if ( view === 'hex' ) {
+			setView( 'rgb' );
+			resetDraftValues();
 
 			speak( __( 'RGB mode active' ) );
-		} else if ( this.state.view === 'rgb' ) {
-			this.setState( { view: 'hsl' }, this.resetDraftValues );
+		} else if ( view === 'rgb' ) {
+			setView( 'hsl' );
+			resetDraftValues();
 
 			speak( __( 'Hue/saturation/lightness mode active' ) );
-		} else if ( this.state.view === 'hsl' ) {
-			if ( this.props.hsl.a === 1 ) {
-				this.setState( { view: 'hex' }, this.resetDraftValues );
+		} else if ( view === 'hsl' ) {
+			if ( hsl.a === 1 ) {
+				setView( 'hex' );
+				resetDraftValues();
 
 				speak( __( 'Hex color mode active' ) );
 			} else {
-				this.setState( { view: 'rgb' }, this.resetDraftValues );
+				setView( 'rgb' );
+				resetDraftValues();
 
 				speak( __( 'RGB mode active' ) );
 			}
 		}
 	}
 
-	resetDraftValues() {
-		return this.props.onChange( {
+	function resetDraftValues() {
+		return onChange( {
 			state: 'reset',
 		} );
 	}
 
-	normalizeValue( valueKey, value ) {
-		if ( valueKey !== 'a' ) {
-			return value;
-		}
-
-		if ( value < 0 ) {
-			return 0;
-		} else if ( value > 1 ) {
-			return 1;
-		}
-		return Math.round( value * 100 ) / 100;
-	}
-
-	handleChange( { source, state, value, valueKey } ) {
-		this.props.onChange( {
+	function handleChange( { source, state, value, valueKey } ) {
+		onChange( {
 			source,
 			state,
 			valueKey,
-			value: this.normalizeValue( valueKey, value ),
+			value: normalizeValue( valueKey, value ),
 		} );
 	}
 
-	renderFields() {
-		const { disableAlpha = false } = this.props;
-		if ( this.state.view === 'hex' ) {
+	function renderFields() {
```

Comment:
I don't like seeing render functions inside functional components (which are also a render function). There's usually a way to rewrite this without needing a function. Here I think you can `let fields` and assign a components based on `view`. It could also be a new component if you prefer (but you have to pass all the used props).

Reply:
`renderFields` function removed in daddou-ma/gutenberg@d4f4aad

Reply:
Nice